test/integration/main_test.go

```diff
@@ -51,12 +53,44 @@ const (
 // TestMain is the test main
 func TestMain(m *testing.M) {
 	flag.Parse()
+	setMaxParallelism()
+
 	start := time.Now()
 	code := m.Run()
 	fmt.Printf("Tests completed in %s (result code %d)\n", time.Since(start), code)
 	os.Exit(code)
 }
 
+// setMaxParallelism caps the max parallelism. Go assumes 1 core per test, whereas minikube needs 2 cores per test.
+func setMaxParallelism() {
+	procs := runtime.GOMAXPROCS(0)
+	pval := flag.Lookup("test.parallel").Value.String()
+	pv, err := strconv.Atoi(pval)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "unable to parse --test.parallel value: %q\n", pval)
+		return
+	}
+
+	if procs != pv {
```

Comment:
was this meant to be if proc < pv {  } ?

Reply:
This is intentional: If someone tests manually --test-parallel, we preserve the value rather than ignoring them..